lib/cli-engine.js

```diff
@@ -300,18 +300,31 @@ function processFile(filename, configHelper, options) {
 
 /**
  * Returns result with warning by ignore settings
- * @param {string} filePath File path of checked code
- * @returns {Result} Result with single warning
+ * @param {string} filePath - File path of checked code
+ * @param {string} baseDir  - Absolute path of base directory
+ * @returns {Result}           Result with single warning
  * @private
  */
-function createIgnoreResult(filePath) {
+function createIgnoreResult(filePath, baseDir) {
+    var message;
+
+    if (/^\./.test(path.basename(filePath))) {
+        message = "Hidden file ignored by default.  Use '--ignore-pattern !.*' to override.";
```

Comment:
Yes, but I would like to avoid using `relativePath` directly in the message because baseDir may not always be provided and there may be other edge cases and conditions that cause trouble.  I'd rather replace all of that logic with:

```
var isHidden = /^\./.test(path.basename(filePath));
var isInNodeModules = baseDir && /^node_modules/.test(path.relative(baseDir, filePath));
var isInBowerComponents = baseDir && /^bower_components/.test(path.relative(baseDir, filePath));

if (isHidden || isInNodeModules || isInBowerComponents) {
  message = "File ignored by default.  Use a negated --ignore-pattern (like '!<relative/path/to/filename>') to override."";
} else {
  message = "File ignored because of a matching ignore pattern. Use '--no-ignore' to override.";
}
```

This is still more complicated than I'd like and still relies on `baseDir` for `node_modules` and `bower_components` (no way around that), but does not require it for hidden files.  The more explicit we are in the error message (by giving an exact flag to use), the greater chances for confusion if/when that suggestion does not work.